File: examples/simple-java/src/main/java/org/teslatoolkit/examples/simple/java/ExampleMain.java
package org.teslatoolkit.examples.simple.java;

import org.teslatoolkit.auth.AccountAuthentication;
import org.teslatoolkit.java.TeslaClientSync;

public class ExampleMain {
  public static void main(String[] args) {
    TeslaClientSync client = TeslaClientSync.create(new AccountAuthentication(args[0], args[1]));

    client
        .listVehicles()
        .forEach(
            (it) -> {
              System.out.println("Vehicle " + it.getVehicleId());
            });
  }
}

File: examples/simple-kotlin/src/main/kotlin/org/teslatoolkit/example/ExampleMain.kt
package org.teslatoolkit.example

import org.teslatoolkit.auth.AccountAuthentication
import org.teslatoolkit.http.TeslaHttpClient

suspend fun main(args: Array<String>) {
  val client = TeslaHttpClient.create(
    AccountAuthentication(
      email = args[0],
      password = args[1]
    )
  )

  client.listVehicles().forEach { vehicle ->
    println("Vehicle ${vehicle.vehicleId}:")
    println("  State: ${vehicle.state}")
  }

  client.close()
}
